gemhardware/managers/src/common/optohybrid/OptoHybridManager.cc

```diff
@@ -262,95 +177,11 @@ void gem::hw::optohybrid::OptoHybridManager::configureAction()
-        CMSGEMOS_INFO("Setting VFAT parameters with broadcast write using mask " << std::hex << vfatMask << std::dec);
-
-        // FIXME OBSOLETE/UPDATE?
-        std::map<std::string, uint16_t> vfatSettings;
-
-        if (m_scanType.value_ == 2) { // FIXME OBSOLETE
-          CMSGEMOS_INFO("OptoHybridManager::configureAction configureAction: FIRST Latency  " << m_scanMin.value_);
-          vfatSettings["Latency"    ] = static_cast<uint16_t>(m_scanMin.value_);
-          // FIXME optohybrid->setVFATsToDefaults(vfatSettings, vfatMask);
-          // HACK
-          // FIXME optohybrid->broadcastWrite("VFATChannels.ChanReg23",  0x40, vfatMask);
-          // FIXME optohybrid->broadcastWrite("VFATChannels.ChanReg124", 0x40, vfatMask);
-          // FIXME optohybrid->broadcastWrite("VFATChannels.ChanReg65",  0x40, vfatMask);
-          // FIXME optohybrid->broadcastWrite("VCal",                    0xaf, vfatMask);
-        } else if (m_scanType.value_ == 3) { // FIXME OBSOLETE
-          uint32_t initialVT1 = m_scanMin.value_;
-          // FIXME uint32_t VT1 = (m_scanMax.value_ - m_scanMin.value_);
-          uint32_t initialVT2 = 0; //std::max(0,(uint32_t)m_scanMax.value_);
-          // FIXME CMSGEMOS_INFO("OptoHybridManager::configureAction FIRST VT1 " << initialVT1 << " VT2 " << initialVT2);
-          // FIXME vfatSettings["VThreshold1"] = (uint8_t)(initialVT1&0xffff);
-          // FIXME vfatSettings["VThreshold2"] = (uint8_t)(initialVT2&0xffff);
-          // FIXME optohybrid->setVFATsToDefaults(vfatSettings, vfatMask);
-        } else {
-          // FIXME optohybrid->setVFATsToDefaults(vfatSettings, vfatMask);
-        }
-
-        CMSGEMOS_INFO("Reading back values after setting defaults:");
+        CMSGEMOS_INFO("OptoHybridManager::configureAction:: configuring VFATs");
         optohybrid->configureVFATs();
-        // auto vfatcfg = optohybrid->readVFATsConfiguration();
-        // for (auto const& cfg : vfatcfg) {
-        //   for (auto const& r : cfg) {
-        //     CMSGEMOS_INFO(" 0x" << std::hex << std::setw(8) << std::setfill('0') << r << std::dec);
-        //   }
-        // }
-
         // what else is required for configuring the OptoHybrid?
         // need to reset optical links?
         // reset counters?
-
-        // FIXME, should not be here or done like this
-        uint32_t gtxMask = optohybrid->readReg("GEM_AMC.DAQ.CONTROL.INPUT_ENABLE_MASK");
-        std::stringstream msg;
-        msg << "OptoHybridManager::configureAction::OptoHybrid connected on link " << static_cast<uint32_t>(link)
-            << " to AMC in slot " << static_cast<uint32_t>(slot+1) << " found, INPUT_ENABLE_MASK changed from "
-            << std::hex << gtxMask << std::dec;
-        gtxMask   |= (0x1<<link);
-        inputMask |= (0x1<<link);
-        // optohybrid->writeReg("GEM_AMC.DAQ.CONTROL.INPUT_ENABLE_MASK", gtxMask);
-        optohybrid->writeReg("GEM_AMC.DAQ.CONTROL.INPUT_ENABLE_MASK", inputMask);
```

Comment:
Where is it done now?

Reply:
nowhere. This should not be done as it was implemented before and should come with a revised configuration methods on the CTP7 side. And certainly should be called on per-OH basis, this is an AMC register, should be set through AMC manager.

Reply:
This is clearly part of the AMC address space and should be treated as such. But, once your PR will be merged, what happens if someone tries to start a run? Will any chamber be readout? How is it set? (I'm not sure I was around at the time "as it was implemented before".)

Reply:
no, the chambers won't be readout, unless the register "GEM_AMC.DAQ.CONTROL.INPUT_ENABLE_MASK" is correctly written (binary mask which enables corresponding chambers). "as before" means each connected OH adds its "1" to the mask and register is written N(OH) times. This register can be set manually at the moment, but I can foresee quite trivial update to daq configuration module of the CTP7 which will take care of it. However this is beyond the scope of the current PR. If you want this functionality in its obsolete form right now, I can add it back.

Reply:
Okay, thank you for the precision. That's fine to me for now. @cgalloni, you might be interested by the behavioral change.